Update scenarios for Consistency05 match updated specification (#1514)
dotse/zonemaster: docs/public/specifications/test-zones/Consistency-TP/consistency05.md

term "out-of-domain" and was used as the negation of "in-bailiwick".

## Test Case
This document specifies defined test zones for test case [Consistency05].

## Test scenarios

The purpose of the test scenarios is to cover all reasonable contexts where
different message tags are outputted when [Consistency05] is run on a test zone.
The message tags are defined in the test case ([Consistency05]) and the scenarios
are defined below.

The test scenarios are structured as stated in the [test scenario README file].

## Test zone names

The test zone for each test scenario in this document is a subdomain delegated
from the base name (`consistency05.xa`) and that subdomain having the same name
as the scenario. For some scenarios the test zone is a child zone to that zone.
The names of those zones are given in section "[Zone setup for test scenarios]"
below.

## All message tags
The test case can output any of these message tags, but not necessarily in any
combination. See [Consistency05] for the specification of the tags.

* CS05_CHILD_ZONE_LAME
* CS05_DELEGATION
* CS05_EXTRA_ADDR_CHILD
* CS05_ID_ADDR_MISMATCH
* CS05_ID_ADDR_MISSING
* CS05_INCONSISTENT_DELEGATION

> * CS05_MISSING_GLUE_FOR_NS
> * CS05_MISSING_GLUE_FOR_NS_UNDEL
> * CS05_NO_MISMATCH_GLUE_ZONE

Reviewer: Just to resync with the specification:
```suggestion
* CS05_MISSING_GLUE_FOR_NS_UNDEL
* CS05_MISSING_GLUE_FOR_ROOT_NS
* CS05_NO_MISMATCH_GLUE_ZONE
```

Author: I will correct.

| ADDRESSES-MATCH-4         | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |
| ADDRESSES-MATCH-5         | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |
| ADDRESSES-MATCH-6         | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |
| ADDRESSES-MATCH-7         | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |
| ADDRESSES-MATCH-8         | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |
| ADDRESSES-MATCH-9         | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |
| CHILD-ZONE-LAME-1         | CS05_CHILD_ZONE_LAME                            | 2)                     |
| CHILD-ZONE-LAME-2         | CS05_CHILD_ZONE_LAME                            | 2)                     |
| CHILD-ZONE-LAME-3         | CS05_CHILD_ZONE_LAME                            | 2)                     |
| EXTRA-ADDRESS-CHILD       | CS05_EXTRA_ADDR_CHILD                           | 2)                     |
| ID-ADDR-MISMATCH-1        | CS05_ID_ADDR_MISMATCH                           | 2)                     |

> | ID-ADDR-MISMATCH-2        | CS05_ID_ADDR_MISMATCH                           | 2)                     |

Reviewer: According to the draft specification (and especially step 13), this scenario will give `CS05_ID_ADDR_MISSING` for ns2, instead of `CS05_ID_ADDR_MISMATCH`.

That’s because `CS05_ID_ADDR_MISSING` is output for each name server name that has glue records in the parent zone, but no address records at all in the child zone.

Author: Correct, and it is redundant. I'll remove it.

> | ID-ADDR-MISMATCH-3        | CS05_ID_ADDR_MISMATCH                           | 2)                     |

Reviewer: Shouldn’t this scenario also output `CS05_ID_ADDR_MISSING` instead of `CS05_ID_ADDR_MISMATCH` for a similar reason?

Author: Correct, I'll change and rename the scenario.

| ID-ADDR-MISSING-1         | CS05_ID_ADDR_MISSING                            | 2)                     |
| INCONSISTENT-DELEGATION-1 | CS05_INCONSISTENT_DELEGATION, CS05_DELEGATION   | 2)                     |
| INCONSISTENT-DELEGATION-2 | CS05_INCONSISTENT_DELEGATION, CS05_DELEGATION   | 2)                     |
| MISSING-GLUE-FOR-NS-1     | CS05_MISSING_GLUE_FOR_NS                        | 2)                     |
| MISSING-GLUE-FOR-NS-2     | CS05_MISSING_GLUE_FOR_NS_UNDEL                  | 2)                     |
| NO-NS-ADDR-CHILD-1        | CS05_NO_NS_ADDR_CHILD, CS05_MISSING_GLUE_FOR_NS | 2)                     |
| NO-NS-ADDR-CHILD-2        | CS05_NO_NS_ADDR_CHILD                           | 2)                     |
| OOD-ADDR-MISMATCH         | CS05_OOD_ADDR_MISMATCH                          | 2)                     |
| ROOT-MATCH-1              | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |

> | ROOT-MISSING-GLUE-UNDEL-1 | CS05_MISSING_GLUE_FOR_NS_UNDEL                  | 2)                     |

Reviewer: ```suggestion
| ROOT-MISSING-GLUE-UNDEL-1 | CS05_MISSING_GLUE_FOR_ROOT_NS                   | 2)                     |
```

| Z-ROOT-MATCH-1            | CS05_NO_MISMATCH_GLUE_ZONE                      | 2)                     |

> | Z-ROOT-INCOMPLETE-HINT    | CS05_MISSING_GLUE_FOR_NS                        | 2)                     |

Reviewer: ```suggestion
| Z-ROOT-INCOMPLETE-HINT    | CS05_MISSING_GLUE_FOR_ROOT_NS                   | 2)                     |
```

Author: Will fix.

  * The format for undelegated data follow the format used for `zonemaster-cli`
    (after `--ns`).

### ADDR-MATCH-DEL-UNDEL-1

> Also the "happy path". But there is an undelegated zone to be tested.

Reviewer: With scenarios having been moved around, this one is the first and it shouldn’t say “*also* the "happy path".

Author: Will fix.

  * Neither ns1 nor ns2 are defined in the zone as address records.
  * The correct NS records are in the zone.

### ID-ADDR-MISSING-2
For ns2 (in-domain), there is no NS for ns2 and the glue does not match any
address records in the zone. Furthermore, ns2 does not respond.

* Zone: id-addr-missing-2.consistency05.xa
  * Normal delegation to ns1 and ns2.
  * There is no NS record for ns2 in the zone.
  * No address records for ns2 (IPv4 or IPv6) in the zone.
  * ns2 does not respond.

### INCONSISTENT-DELEGATION-1
Delegation from parent zone is inconsistent.

* Zone: child.inconsistent-delegation-1.consistency05.xa
  * Parent zone is inconsistent-delegation-1.consistency05.xa
    * Parent ns1 delegates to child ns41 and ns42
    * Parent ns2 delegates to child ns41 and ns43
  * Child ns are in-domain
    * ns41, ns42 and n43
    * Child is consistent

### INCONSISTENT-DELEGATION-2
Delegation from parent zone is inconsistent.

* Zone: child.inconsistent-delegation-2.consistency05.xa
  * Parent zone is inconsistent-delegation-2.consistency05.xa
    * Parent ns1 delegates to child ns41 and ns42
    * Parent ns2 delegates to child ns41 and ns43
  * Child ns are out-of-domain

>     * ns41, ns42 and n43

Reviewer: ```suggestion
    * ns41, ns42 and ns43
```

    (IPv4 or IPv6) in the zone.

### ID-ADDR-MISMATCH-3
For ns2 (in-domain), there is no NS for ns2 and the glue does not match any
address records in the zone. Furthermore, ns2 does not respond.

* Zone: id-addr-mismatch-3.consistency05.xa
  * Normal delegation to ns1 and ns2.
  * There is no NS record for ns2 in the zone.
  * No address records for ns2 (IPv4 or IPv6) in the zone.
  * ns2 does not respond.

### ID-ADDR-MISSING-1
Both NS are in-domain and exist with correct glue in the delegation, but there
are no address records in the zone matching the name server names.

* Zone: id-addr-missing-1.consistency05.xa
  * Neither ns1 nor ns2 are defined in the zone as address records.
  * The correct NS records are in the zone.

### INCONSISTENT-DELEGATION-1
Delegation from parent zone is inconsistent.

* Zone: child.inconsistent-delegation-1.consistency05.xa
  * Parent zone is inconsistent-delegation-1.consistency05.xa
    * Parent ns1 delegates to child ns41 and ns42
    * Parent ns2 delegates to child ns41 and ns43
  * Child ns are in-domain
    * ns41, ns42 and n43
    * Child is consistent

### INCONSISTENT-DELEGATION-2
Delegation from parent zone is inconsistent.

> * Zone: child.inconsistent-delegation-2.consistency05.xa
>   * Parent zone is inconsistent-delegation-2.consistency05.xa
>     * Parent ns1 delegates to child ns1 and ns2
>     * Parent ns2 delegates to child ns1 and ns3
>   * Child ns are out-of-domain
>     * ns1, ns2 and n3
>     * Child is consistent

Reviewer: ```suggestion
* Zone: child.inconsistent-delegation-2.consistency05.xa
  * Parent zone is inconsistent-delegation-2.consistency05.xa
    * Parent ns1 delegates to child ns41 and ns42
    * Parent ns2 delegates to child ns41 and ns43
  * Child ns are out-of-domain
    * ns41, ns42 and ns43
    * Child is consistent
```

Author: Will update.

### MISSING-GLUE-FOR-NS-1
Delegation lacks mandatory glue for ns41.

* Zone: child.missing-glue-for-ns-1.consistency05.xa
  * Parent ns1 and ns2 have the same delegation of child.
  * Child runs on ns41 and ns42.
  * Delegation lacks glue for ns41.

### MISSING-GLUE-FOR-NS-2
The child zone is undelegated with undelegated data. Undelegated data lacks
mandatory glue for ns1.

* Zone: missing-glue-for-ns-2.consistency05.xa
  * No delegated zone.
  * Undelegated zone on ns1 and ns2.
  * Glue is missing for ns1 in undelegated data.
  * Undelegated data:
    * ns1.missing-glue-for-ns-2.consistency05.xa

>     * ns1.missing-glue-for-ns-2.consistency05.xa

Reviewer: ```suggestion
    * ns1.missing-glue-for-ns-2.consistency05.xa
```

Author: Will fix.

Lame. There are no IP addresses to the child zone

* Zone: child.no-ns-addr-child-2.consistency05.xa
  * ns1 and ns2 are out-of-domain.
  * No glue in delegation.
  * Authoriative records of ns1 and ns2 do not exist.
  * Child zone does not have to be created.

### OOD-ADDR-MISMATCH
For one NS (out-of-domain, but with glue) glue does not match AA address
response.

* Zone: child.ood-addr-mismatch.consistency05.xa
  * Both ns1 and ns2 are out-of-domain
  * ns1 is "ns1.sibbling.ood-addr-mismatch.consistency05.xa"
  * ns2 is "ns2.sibbling.ood-addr-mismatch.consistency05.xa"
  * Delegation is with glue.
  * The "sibling" zone has full set of address records
  * ns1 in the "sibling" zone matches the addresses of glue.
  * ns2 in the "sibling" zone does not match the addresses of glue.
  * All IP addresses of ns1 and ns2 must serve identical versions of the zone.

### ROOT-MATCH-1
Default root zone and hintfile, and all matches.

* Zone: . (root)
  * Dedicated hintfile not needed.

### ROOT-MISSING-GLUE-UNDEL-1
Default root zone and hintfile. Undelegated is used for another root zone. Glue
is missing.

* Zone: . (root)
  * Dedicated hintfile not needed.
  * Undelegated data:
    * ns1

>     * ns1

Reviewer: ```suggestion
    * ns1
```

Author: Will fix.
